packages/flutter/test/material/date_range_picker_test.dart

```diff
@@ -404,6 +406,71 @@ void main() {
     });
   });
 
+  testWidgets('Can select a range even if the range includes non selectable days', (WidgetTester tester) async {
+    await preparePicker(tester, (Future<DateTimeRange?> range) async {
+      await tester.tap(find.text('12').first);
+      await tester.tap(find.text('14').first);
+      await tester.tap(find.text('SAVE'));
+      // The day 13 is not selectable, but the range is still valid.
+      expect(await range, DateTimeRange(
+        start: DateTime(2016, DateTime.january, 12),
+        end: DateTime(2016, DateTime.january, 14),
+      ));
+    }, selectableDayPredicate: (DateTime day, _, __) => day.day != 13);
+  });
+
+  testWidgets('Cannot select a day inside bounds but not selectable', (WidgetTester tester) async {
+    initialDateRange = DateTimeRange(
+      start: DateTime(2017, DateTime.january, 13),
+      end: DateTime(2017, DateTime.january, 14),
+    );
+    firstDate = DateTime(2017, DateTime.january, 12);
+    lastDate = DateTime(2017, DateTime.january, 16);
+    await preparePicker(tester, (Future<DateTimeRange?> range) async {
+      // non selectable date. Should be ignored.
+      await tester.tap(find.text('15'));
+      await tester.tap(find.text('SAVE'));
+      // We should still be on the initial date.
+      expect(await range, initialDateRange);
+    }, selectableDayPredicate: (DateTime day, _, __) => day.day != 15);
+  });
+
+  testWidgets('Selectable date becoming non selectable when selected start day', (WidgetTester tester) async {
+    await preparePicker(tester, (Future<DateTimeRange?> range) async {
+      await tester.tap(find.text('12').first);
+      await tester.pumpAndSettle();
```

Comment:
Try to minimize the use of `pumpAndSettle`. Is there a specific duration for the animation that can be used? Is there even an animation happening that requires more than `pump`? Just making sure you've explored those.

Reply:
Yes, there is an animation to wait apparently, I suppose the one when clicking. It's not working with just `pump`